### ISSUE TITLE
cmake 2.8.x does not like <TARGET_FILE:tgt> expressions in CMakeLists.txt

### ISSUE THREAD

**Issue by nyetwurk** [symless/synergy#5690]
Commit d48f6801f0a8ef79845a5b7025f2257f7e7e1900 breaks cmake 2.8.x

The <TARGET_FILE:tgt> expression does not expand. Haven't tried under cmake 3 but I assume that is the problem.

**Comment by nyetwurk** [comment]
"D:\home\nyet\src\synergy\build\synergy.sln" (default target) (1) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergyc\synergyc.vcxproj.metaproj" (default target) (33) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergyc\synergyc.vcxproj" (default target) (34) ->
(PostBuildEvent target) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: The command "mt.exe -manifest "D:/home/nyet/src/synergy\res\dpiaware.manifest" -inputresource:"$<TARGET_FILE:
synergyc>";#1 -outputresource:"$<TARGET_FILE:synergyc>";#1\r [D:\home\nyet\src\synergy\build\src\cmd\synergyc\synergyc.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: :VCEnd" exited with code 31. [D:\home\nyet\src\synergy\build\src\cmd\synergyc\synergyc.vcxproj]


"D:\home\nyet\src\synergy\build\synergy.sln" (default target) (1) ->
```
"D:\home\nyet\src\synergy\build\src\cmd\synergyd\synergyd.vcxproj.metaproj" (default target) (35) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergyd\synergyd.vcxproj" (default target) (36) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: The command "mt.exe -manifest "D:/home/nyet/src/synergy\res\dpiaware.manifest" -inputresource:"$<TARGET_FILE:
synergyd>";#1 -outputresource:"$<TARGET_FILE:synergyd>";#1\r [D:\home\nyet\src\synergy\build\src\cmd\synergyd\synergyd.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: :VCEnd" exited with code 31. [D:\home\nyet\src\synergy\build\src\cmd\synergyd\synergyd.vcxproj]


"D:\home\nyet\src\synergy\build\synergy.sln" (default target) (1) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj.metaproj" (default target) (39) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj" (default target) (40) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: The command "mt.exe -manifest "D:/home/nyet/src/synergy\res\dpiaware.manifest" -inputresource:"$<TARGET_FILE:
synergys>";#1 -outputresource:"$<TARGET_FILE:synergys>";#1\r [D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.CppCommon.targets(113,5): error MSB3073: :VCEnd" exited with code 31. [D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj]
```

**Comment by nyetwurk** [comment]
Looks like cmake 3.x works.

Should probably update https://github.com/symless/synergy/wiki/Compiling#Dependencies accordingly.

**Comment by nyetwurk** [comment]
```
"D:\home\nyet\src\synergy\build\synergy.sln" (default target) (1) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj.metaproj" (default target) (39) ->
"D:\home\nyet\src\synergy\build\src\cmd\synergys\synergys.vcxproj" (default target) (40) ->
  D:/home/nyet/src/synergy\res\dpiaware.manifest : manifest authoring warning 81010002: Unrecognized Element "compatibility" in namespace "urn:schemas-microsoft-com:compatibility.v1". [D:\home\nyet\src\synergy\bu
ild\src\cmd\synergys\synergys.vcxproj]

D:\home\nyet\src\synergy>mt /?
Microsoft (R) Manifest Tool version 6.3.9600.17298
Copyright (c) Microsoft Corporation 2012.
All rights reserved
````

**Comment by nyetwurk** [comment]
Sigh
https://support.microsoft.com/en-us/kb/2670561

**Comment by nyetwurk** [comment]
Perhaps persist_cmake() in commands1.py needs to check for version 3.0.2 or greater?

**Comment by nlyan** [comment]
@nyetwurk the download for this Hotfix is here http://hotfixv4.microsoft.com/Visual%20Studio%202010/sp1/DevDiv956672/40219.392/free/446240_intl_i386_zip.exe

**Comment by nyetwurk** [comment]
The hotfix is a separate issue from TARGET_FILE

**Comment by XinyuHou** [comment]
Reverted all the changes. The fix for #5186 will make this redundant.
